app/scripts/controllers/createSecret.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateSecretController
+ * @description
+ * # CreateSecretController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('CreateSecretController', function ($scope, $filter, DataService, AlertMessageService) {
+
+    $scope.newSecret = {
+      type: $scope.type === "Git Secret" ? "Source Secret" : $scope.type,
```

Comment:
Well it depends on the type of the secret that user wants to create. If he clicks on the create `Create Pull Secret` we would prefill the create modal with all the fields for creating that secret.